codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerProtocolHandshakeHandler.java

```diff
@@ -110,6 +105,10 @@ public void operationComplete(ChannelFuture future) throws Exception {
         }
     }
 
+    private boolean isNotWebSocketPath(FullHttpRequest req) {
+        return checkStartsWith ? !req.uri().startsWith(websocketPath) : !req.uri().equals(websocketPath);
```

Comment:
Your code currently has a branch statement: `if <condition> then <statement> else <statement>` however the suggestion above removes a branch at the expense of evaluating the same boolean twice. Although the difference is likely in the noise there is a possibility these two will have different characteristics.